src/components/Guides/PlaylistPage.tsx

```diff
@@ -1,35 +1,115 @@
 import React from 'react';
+import { Typography } from '@material-ui/core';
+import { withStyles, Theme } from '@material-ui/core/styles';
+import { styles } from './PlaylistStyles';
 import { useSelector } from 'react-redux';
 import { RootState } from '../../store/reducers';
 import StandardCard from '../Cards/StandardCard';
+import { ReactComponent as InstructionIcon } from './InstructionIcon.svg';
+import { ReactComponent as VideoIcon } from './VideoIcon.svg';
 
 type PlaylistPageProps = {
-  lessonId: string,
-}
+  lessonId: string;
+  classes: any;
+};
 
-const PlaylistPage = ({ lessonId }: PlaylistPageProps) => {
+const PlaylistPage = ({ lessonId, classes }: PlaylistPageProps) => {
   const lessons = useSelector((state: RootState) => state.lessons);
   const resources = useSelector((state: RootState) => state.resources);
+
   const currentLesson = lessons[lessonId];
+  const countMedia = (media: string) =>
+    currentLesson.resourceIDs.filter((id) => resources[id].type == media)
+      .length;
+
+  const calcProgress = () => {
+    const numCompleted = currentLesson.resourceIDs.filter(
+      (id) => resources[id].isCompleted
+    ).length;
+    return Math.round(
+      ((numCompleted * 1.0) / currentLesson.resourceIDs.length) * 100
+    );
+  };
+
+  const barStyles = {
+    width: '200px',
+    height: '5px',
+    background: 'white',
+    boxShadow: '-2px 4px 20px rgba(79, 42, 29, 0.15)',
+    borderRadius: '20px',
+    marginRight: '0.3rem'
+  };
+
+  const fillStyles = {
+    width: `${calcProgress()}%`,
+    height: '5px',
+    background: '#33BF68',
+    borderRadius: '20px'
+  };
+
+  const ProgressBar = () => {
```

Comment:
Oo, could we potentially move this progressbar into another component? :0

Reply:
done :)